d3ploy/demand_driven_deployment_inst.py

```diff
@@ -219,17 +241,28 @@ def decision(self):
         """
         time = self.context.time
         for commod, proto_dict in self.commodity_dict.items():
-
             diff, supply, demand = self.calc_diff(commod, time)
             lib.record_time_series('calc_supply' + commod, self, supply)
             lib.record_time_series('calc_demand' + commod, self, demand)
-
             if diff < 0:
-                deploy_dict = solver.deploy_solver(
-                    self.commodity_supply, self.commodity_dict, commod, diff, time)
+                if self.installed_cap:
+                    deploy_dict = solver.deploy_solver(
+                        self.installed_capacity, self.commodity_dict, commod, diff, time)
+                else:
+                    deploy_dict = solver.deploy_solver(
+                        self.commodity_supply, self.commodity_dict, commod, diff, time)
                 for proto, num in deploy_dict.items():
                     for i in range(num):
                         self.context.schedule_build(self, proto)
+                # update installed capacity dict
+                for proto, num in deploy_dict.items():
+                    self.installed_capacity[commod][time + 1] = \
+                        self.installed_capacity[commod][time] + \
+                        self.commodity_dict[commod][proto]['cap'] * num
+            else:
```

Comment:
rather than this extra `if diff<0:...  else:` layer.... 

... couldn't the default num (returned by the solver) be 0 for any prototypes that don't need to be deployed?

Reply:
I agree, seems like this would be the smart course of action.